src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java

```diff
@@ -415,7 +416,11 @@ public int compareTo(LocalizedMessage other) {
 
         if (lineNo == other.lineNo) {
             if (columnNo == other.columnNo) {
-                result = getMessage().compareTo(other.getMessage());
+                result = Comparator.nullsFirst(Comparator.<String>naturalOrder())
```

Comment:
@Luolc , please add such classes to known list - https://github.com/checkstyle/checkstyle/blob/master/config/import-control.xml#L19 to disallow usage of them till we are on cobertura. Such update could be separate PR, to avoid mixing too much in one PR.